### PR TITLE
layers: Initial VK_KHR_portability_subset support

### PR DESCRIPTION
**Pull request by ncesario-lunarg**
Add checks for the following VUIDs:
- VUID-VkDeviceCreateInfo-pProperties-04451.
- VUID-vkCreateEvent-events-04468.

Change-Id: I61743dc7ceddd2f4e627a7caeafacfcd0ff19d77

### CODE REVIEW

File: layers/parameter_validation_utils.cpp [hunk 1]
```diff
@@ -286,12 +286,27 @@ bool StatelessValidation::manual_PreCallValidateCreateDevice(VkPhysicalDevice ph
                                 "VUID-VkDeviceCreateInfo-ppEnabledLayerNames-parameter", pCreateInfo->ppEnabledLayerNames[i]);
     }
 
+    // If this device supports VK_KHR_portability_subset, it must be enabled
+    const std::string portability_extension_name("VK_KHR_portability_subset");
+    const auto &dev_extensions = device_extensions_enumerated.at(physicalDevice);
+    const bool portability_supported = dev_extensions.count(portability_extension_name) != 0;
+    bool portability_requested = false;
```
Comment (ncesario-lunarg) [inline layers/parameter_validation_utils.cpp:293]: There is an occasional crash on the 32b internal Windows AMD build agent when running tests. This is the only code that will be getting run in the "general" case (i.e., when portability is not enabled, which is how the tests are currently run).

**Review by mark-lunarg — APPROVED** [review]

**Comment by mark-lunarg** [comment]
Looks great!  Sorry I didn't get to it sooner.  Is the 32-bit issue sorted?

**Comment by ncesario-lunarg** [comment]
Thanks for taking a look at this @mark-lunarg. This is still causing sporadic crashes on the AMD machine, and it appears to have nothing to do with 32 vs 64 bit, but rather a crash is "likely to happen" _after_ any of the portability tests run. Do you see anything suspicious in the way I'm checking to see whether or not a test should run? e.g., this should be the only code that gets run for each portability test (since they won't be supported on any of the build agents currently):
```cpp
    TEST_DESCRIPTION("Portability: CreateDevice called and VK_KHR_portability_subset not enabled");

    ASSERT_NO_FATAL_FAILURE(InitPortabilitySubsetFramework());

    bool portability_supported = DeviceExtensionSupported(gpu(), nullptr, VK_KHR_PORTABILITY_SUBSET_EXTENSION_NAME);
    if (!portability_supported) {
        printf("%s Test requires VK_KHR_portability_subset, skipping\n", kSkipPrefix);
        return;
    }
```
If I make the tests a compile time option and simply don't include them by default, everything seems stable on the AMD machine, so this would seem to indicate I'm doing something wonky with the tests.

**Comment by ncesario-lunarg** [comment]
@mark-lunarg I've removed the setup in the test code that was causing issues on the Windows AMD build agent and updated the documentation as this now requires explicitly adding the devsim layer when running the portability tests through devsim (which might be a better setup anyway).

**Pull request merged**